server/src/controllers/analyticsController.ts

```diff
@@ -0,0 +1,31 @@
+import { StatusCode } from "../enums/StatusCode";
+import { AnalyticsService } from "../services/analyticsService";
+import { Request, Response } from "express";
+
+export class AnalyticsController {
+  static async getAllCheckInUsers(req: Request, res: Response) {
+    try {
+      const allCheckInUsers = await AnalyticsService.getAllCheckInUsers();
+
+      res.send(allCheckInUsers);
+    } catch (error) {
+      res.status(StatusCode.NOT_FOUND).send({ message: error });
+    }
+  }
+
+  static async checkOffUser(req: Request, res: Response) {
+    const { id } = req.params;
+
+    try {
+      const checkedUser = await AnalyticsService.checkOffuser(id);
+
+      if (!checkedUser) {
+        res.status(StatusCode.NOT_FOUND).send({ message: `user not found` });
```

Comment:
add a return statement here. if the code reaches here it will attempt to send response and then it will continue to the next line and attempt to send another response causing an error and the db to fail